### PR TITLE
Feat: add possibility of full initialization if web inspector. 

### PR DESCRIPTION
**Pull request by imurchie**
Much of this is refactoring to organize the messages. And then add the possibility to do every step that desktop safari does when you start a web inspector debugging session. This is turned off by default. 

Once this is in I have some work to simplify things.

### CODE REVIEW

File: lib/remote-messages.js [hunk 1]
```diff
@@ -206,100 +210,199 @@ class RemoteMessages {
     let method, params;
 
     switch (command) {
+      /* BASIC COMMANDS */
       case 'setConnectionKey':
         return this.setConnectionKey({connId});
       case 'indicateWebView':
-        return this.indicateWebView({connId, appIdKey, pageIdKey, opts});
+        return this.indicateWebView({connId, appIdKey, pageIdKey, enabled: opts.enabled});
```
Comment (mykola-mokhnach) [inline lib/remote-messages.js]: can we unpack `enabled` above?

File: lib/rpc-client.js [hunk 2]
```diff
@@ -310,26 +316,129 @@ export default class RpcClient {
...
 
+    // go through the steps that the Desktop Safari system
+    // goes through to initialize the Web Inspector session
+
     const sendOpts = {
       appIdKey,
       pageIdKey,
     };
 
+    for (const enabled of [true, false]) {
```
Comment (mykola-mokhnach) [inline lib/rpc-client.js]: what this magic loop is doing?

File: lib/rpc-client.js [hunk 3]
```diff
@@ -310,26 +316,129 @@ export default class RpcClient {
...
+      shouldPause: false,
+    }, false);
+
+    await this.send('LayerTree.enable', sendOpts, false);
+    await this.send('Worker.enable', sendOpts, false);
+    await this.send('Canvas.enable', sendOpts, false);
+    await this.send('Console.enable', sendOpts, false);
+    await this.send('DOM.getDocument', sendOpts, false);
+    const loggingChannels = await this.send('Console.getLoggingChannels', sendOpts);
+    for (const source of (loggingChannels.channels || {}).map((entry) => entry.source)) {
```
Comment (mykola-mokhnach) [inline lib/rpc-client.js]: can `map` be applied to an object?

**Comment by mykola-mokhnach** [comment]
tests, probably, need some updates

**Comment by imurchie** [comment]
No need to update the tests. This is all refactoring other than adding a flag to do the I initialization.

File: lib/rpc-client.js (hunk 2 above)
Comment (imurchie) [reply lib/rpc-client.js]: Sending a `true` then a `false` to the same command. This highlights and then unhighlights the webview.

File: lib/rpc-client.js (hunk 3 above)
Comment (imurchie) [reply lib/rpc-client.js]: Nope. Fixing.

File: lib/rpc-client.js (hunk 2 above)
Comment (mykola-mokhnach) [reply lib/rpc-client.js]: would be nice to have some comment there

**Pull request merged**